fileuploader-kiwiirc-plugin/components/fileListComponent.vue

```diff
@@ -0,0 +1,140 @@
+            <table class="kiwi-filebuffer-table">
+                <tr><th>Nick</th><th>Link</th><th>Time</th></tr>
+                <tr v-for="(file, idx) in fileList" :key="file">
+                    <td class="kiwi-filebuffer-table-td">
+                        {{file.nick}}
+                    </td>
+                    <td class="kiwi-filebuffer-table-td">
+                        <a
+                            :href="file.url"
+                            @click.prevent.stop="loadContent(file.url)"
+                            class="kiwi-filebuffer-anchor"
+                        >
+                            {{getFileName(file.url)}}
+                        </a>
+                    </td>
+                    <td class="kiwi-filebuffer-table-td">
+                        {{file.time}}
+                    </td>
+                </tr>
+            </table>
+        </div>
+    </div>
+</template>
+
+<script>
+
+'kiwi public';
+
+export default {
+    data() {
+        return {
+            fileList: [],
+            settings: kiwi.state.setting('fileuploader'),
+        };
+    },
+    methods: {
+        getFileName(file) {
+            let name = file.split('/')[file.split('/').length-1];
+            if (name.length >= 25) {
+                name = name.substring(0, 18) + '...' + name.substring(name.length - 4);
+            }
+            return name;
+        },
+        loadContent(url) {
+            kiwi.emit('mediaviewer.show', url);
+        },
+        messageHandler() {
+            setTimeout(() => this.sharedFiles(kiwi.state.getActiveBuffer()), 1100)
+        },
+        pruneDups(a) {
+            for(let i = 0; i < a.length; ++i) {
+                for(let j = i + 1; j < a.length; ++j) {
+                    if(a[i].url === a[j].url) a.splice(j, 1)
+                }
+            }
+        },
+        sharedFiles(buffer) {
+            this.fileList = []
+            for(let i = 0; i < buffer.messagesObj.messages.length; i++) {
+                let e = buffer.messagesObj.messages[i]
+                if (e.message.indexOf(this.settings.server) !== -1) {
+                    let currentdate = new Date(e.time);
+                    let time = ("00" + currentdate.getHours()).slice(-2) + ":"
+                        + ("00" + currentdate.getMinutes()).slice(-2) + ":"
+                        + ("00" + currentdate.getSeconds()).slice(-2)
+                    let link = {
+                        url: e.message.substring(e.message.indexOf(this.settings.server)),
+                        nick: e.nick,
+                        time
+                    };
+                    link.url = link.url.split(' ')[0].split(')')[0]
+                    this.fileList.push(link);
+                }
+            }
+            // comment out the following line to include duplicates
+            this.pruneDups(this.fileList)
+            kiwi.emit('files.listshared', { fileList: this.fileList, buffer })
+            return this.fileList
```

Comment:
This function returns `this.fileList` while also having the side effect of editing `this.fileList` too. It would be cleaner if this created a new array and returned that, instead of editing the component property. Perhaps move the `fileList` data property into a computed property which returns the value of `sharedFiles(kiwi.state.getActiveBuffer())`

Reply:
this one was tricky. thanks again for help.